functions/openhabian.bash

```diff
@@ -63,38 +63,49 @@ openhabian_console_check() {
 
   warningText="We detected that you use a console which is less than 120 columns wide. This tool is designed for a minimum of 120 columns and therefore some menus may not be presented correctly. Please increase the width of your console and rerun this tool.\\n\\nEither resize the window or consult the preferences of your console application."
 
-  whiptail --title "Compatibility Warning" --msgbox "$warningText" 13 80
+  whiptail --title "Compatibility warning" --msgbox "$warningText" 13 80
 }
 
 
-## Check against openhabian.conf for need to update parameters
+## Store a set parameter into openhabian.conf
+## Parameter to write into openhabian.conf
+##
+##    store_in_conf(String parameter)
+##
+store_in_conf() {
+  if [[ -z "$1" ]]; then return 0; fi
+  # shellcheck disable=SC2154
+  sed -i "s|^${1}=.*$|$1=${!1}|g" "$configFile"
+} 
+
+
+## Check openhabian.conf against openhabian.conf.dist for need to add parameters
 ## Add a parameter if it's missing or keep it if user provided.
 ## Cycling through openhabian.conf.dist will ensure all current and future params are checked for
-## Run on every start (to ensure params are *up to date* whenever the user changes branch or openhabian-config unattended is run)
+## Run on every start (to ensure params are up to date at any time)
 ##
 ##    update_openhabian_conf()
 ##
 update_openhabian_conf() {
-  local config=/etc/openhabian.conf
+  local configFile=/etc/openhabian.conf
   local referenceConfig=/opt/openhabian/openhabian.conf.dist
 
-  cp $config ${config}.BAK
-  cp /dev/null $config
-
+  cp $configFile ${configFile}.BAK
   while read -r line; do
-    if [[ $line =~ ^# ]] || [[ -z "$line" ]]; then  # if line is a comment or empty
-      echo "$line"
-    else                                            # not fully sure if anything else is a proper param setting but let's assume this
-      param=$(echo "$line" | cut -d'=' -f1)         # get parameter name first
-      if [[ -z ${!param+x} ]]; then                 # if $param is set (if so it is because it was sourced on start)
-        echo "$line"                                # take the param from the default config by printing the default .conf's full line
-      elif [[ ${!param} == *" "* ]]; then
-      	echo "$param=\"${!param}\""                 # else parameter is set i.e. a line setting it is present in $config so use its value
+    if [[ $line =~ ^[a-zA-Z] ]]; then         # if line is a comment or empty
+      param=$(echo "$line" | cut -d'=' -f1)   # get parameter name first
```

Comment:
Why are you changing this, if the line is a comment should we not just copy it over right away?

Reply:
I fully redid the function. The comment is bad - changed that.